addons/mail/static/src/core/common/thread_actions.js

```diff
@@ -56,7 +56,8 @@ function transformAction(component, id, action) {
         /** Closes this action. */
         close() {
             if (this.toggle) {
-                component.threadActions.activeAction = null;
+                component.threadActions.activeAction =
+                    component.threadActions.actionStack.pop() ?? null;
```

Comment:
Value `null` is not really important. As long it's falsy it's good, and `pop()` returns `undefined` when no items in stack, which is good enough